core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -98,9 +101,11 @@ public function createContaoWebpageResponseContext(PageModel $pageModel): Respon
 
             // Ensure absolute links
             if (!preg_match('#^https?://#', $url)) {
-                /** @var Environment $environment */
-                $environment = $this->contaoFramework->getAdapter(Environment::class);
-                $url = $environment->get('base').$url;
+                if (!$request = $this->requestStack->getMainRequest()) {
+                    throw new \RuntimeException('The request stack did not contain a request');
+                }
+
+                $url = $request->getSchemeAndHttpHost().'/'.$url;
```

Comment:
In case you wonder, we do not need `$request->getBaseUrl()` here, because the base URL (e.g. `preview.php`) is already added when the insert tag is replaced.